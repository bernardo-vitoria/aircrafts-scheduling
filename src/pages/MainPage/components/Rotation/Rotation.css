.rotation {
    overflow-x: hidden;
    overflow-y: auto;
    height: 84vh;
}
.rotation-title {
    padding: 8px;
    text-align: center;
    padding-bottom: 4px;
};