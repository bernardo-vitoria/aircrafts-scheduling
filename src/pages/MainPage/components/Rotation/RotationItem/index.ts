import RotationItem from "./RotationItem";

export { RotationItem };