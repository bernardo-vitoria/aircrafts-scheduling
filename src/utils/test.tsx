// @ts-nocheck
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { I18nextProvider } from "react-i18next";
import { QueryClient, QueryClientProvider } from "react-query";
import { MemoryRouter } from "react-router-dom"; // 4.0.0
import { queries } from "@testing-library/react";
import {
  render,
  RenderResult,
} from "@testing-library/react";
import i18n from "../config/i18nForTests";

const customRender = (ui: React.ReactElement, options = {}): RenderResult => {
  const CombinedProviders: React.FC = ({ children }) => {
    const config = {
      queries: {
        retry: false,
        cacheTime: 0,
        refetchOnWindowFocus: false,
        retryDelay: 0,
      },
    };

    const queryClient = new QueryClient({ defaultOptions: config });

    return (
        <QueryClientProvider client={queryClient}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter>{children}</MemoryRouter>
          </I18nextProvider>
        </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: CombinedProviders, queries, ...options });
};

export {
  customRender as render
};