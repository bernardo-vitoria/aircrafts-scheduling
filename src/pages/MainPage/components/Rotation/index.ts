import Rotation from './Rotation';

export { Rotation };