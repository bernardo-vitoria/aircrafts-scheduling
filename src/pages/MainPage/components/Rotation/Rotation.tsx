import React from 'react';
import { RotationItem } from './RotationItem';
import { Paper, Stack } from '../../../../components';
import { useRotationStack } from '../../../../store';
import { Flight } from '../../types';
import "./Rotation.css";

const Rotation: React.FC = () => {
    const { selectedAircraft, flightsRotation } = useRotationStack((state) => ({
      selectedAircraft: state.selectedAircraft,
      flightsRotation: state.flightsRotation,
    }));

    return (
      <Paper sx={{ height: "90vh", width: "60vh" }}>
        <div className="rotation-title">
          Rotation {selectedAircraft && <b>{selectedAircraft.ident}</b>}
        </div>
        <Stack
          direction="column"
          justifyContent="space-evenly"
          alignItems="center"
          spacing={2}
        >
          <div className="rotation">
            {flightsRotation.map((flight: Flight) => (
              <RotationItem flight={flight} key={`rotationItem-${flight.id}`} />
            ))}
          </div>
        </Stack>
      </Paper>
    );
};

export default Rotation;