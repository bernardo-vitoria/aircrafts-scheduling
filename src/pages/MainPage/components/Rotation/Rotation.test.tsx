import Rotation from "./Rotation";
import { render } from "../../../../utils/test";

describe("Rotation", () => {
  it("should match snapshot", () => {
    const { container } = render(<Rotation />);

    expect(container).toMatchSnapshot();
  });
});
