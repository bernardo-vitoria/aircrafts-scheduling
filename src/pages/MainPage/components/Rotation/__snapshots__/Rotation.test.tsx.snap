// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rotation should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-9j59e7-MuiPaper-root"
  >
    <div
      class="rotation-title"
    >
      Rotation 
    </div>
    <div
      class="css-1lgskws-MuiStack-root"
    >
      <div
        class="rotation"
      />
    </div>
  </div>
</div>
`;
